### DISCUSSION TITLE
Starship can't read its config file

### DISCUSSION THREAD

**Discussion by jarmos-san** [git-for-windows/git#3362] [Q&A] [answered]
I use [Starship](https://starship.rs/) to customize my prompt(s). As [suggested in the docs](https://starship.rs/config/#configuration), I have it configured through a custom config file as well which is located under `%USERPROFILE%`. There's no issue on Windows PowerShell whatsoever but it fails on Git Bash due to permission issues.

So, each time I start a Git Bash session, I see this error:

```console
[ERROR] - (starship::config): Unable to read config file content: Access is denied. (os error 5)
```

I've tried invoking a Git Bash session with admin privileges as well but with no luck. So, is there anything I should try out to debug this issue?

Or if there's a solution to it, it would be nice if you could share it.

**Comment by rimrul** [comment]
Looking at their documentation, they seem to be trying to open `~/.config/starship.toml` (unless something else is configured in `$STARSHIP_CONFIG`). So I'd probably first take a look at the content of `$HOME` and `$STARSHIP_CONFIG`.

**Reply by jarmos-san** [reply]
Hmm...weird! I don't see the error message anymore & I might've resolved this issue with very subtle fix (_not sure if I should call it that_) as well lol

**Here's what worked**:

```bash
export $STARSHIP_CONFIG="$HOME/path/to/config/file"
```

**What didn't work is**:

```bash
export $STARSHIP_CONFIG="~/path/to/config/file"
```

The `~` & the `$HOME` made all the difference, even though they should evaluate to the same value.

Any idea what's going on here? Is it an issue with `git-for-windows` or `starship`? If its the later than I think report this issue on its bug tracker.

**Reply by rimrul** [reply]
Ok, I think I know what's going on here.
`$HOME` probably gets expanded by bash as soon as you export `$STARSHIP_CONFIG` so that probably then gets converted from a value like `/c/users/foo/path/to/config` to something like `c:/users/foo/path/to/config` when calling the native starship executable.

The version with `~` on the other hand? Bash expects the recieving application to expand that. Starship probably passes the unmodified value to the native function to open a file and Windows being Windows doesn't find a folder named `~`. So not really a bug in either one, but an odd side effect from crossing between unix-ish and non-unix-ish systems.

**Comment by jarmos-san** [comment, marked as answer]
For anyone else stumbling across this issue, here's what you got to do:

Leave this line of code in your `.bashrc`:

```bash
# Example: $HOME/.config/starship.toml
export $STARSHIP_CONFIG="$HOME/path/to/config/file"
```

Refer to [this thread](https://github.com/git-for-windows/git/discussions/3362#discussioncomment-1184299) for a possible explanation on this issue. And for more info on configuration details find the details on [the official docs](https://starship.rs/config/). Additionally, feel free to track this issue [here](https://github.com/starship/starship/issues/2986).
